src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,167 @@
+                    return $methods[$methodClass];
+                }, $registeredMethods);
+                $registeredMethodList = implode(', ', $registeredMethodNames);
+                $methodCount = count($registeredMethods);
+                $member->methodCount = $methodCount ?: '';
+                $member->registeredMethods = $registeredMethodList;
+                $member->defaultMethodName = $methods[$defaultMethodClassName] ?? '';
+            }
+            $member->skippedRegistration = $member->HasSkippedMFARegistration
+                ? $member->dbObject('HasSkippedMFARegistration')->Nice()
+                : '';
+        });
+        $filteredList = ArrayList::create($members);
+        if (isset($countFilter)) {
+            $filteredList = $filteredList->filter('methodCount', $countFilter);
+        }
+        return $filteredList;
+    }
+
+    public function columns()
+    {
+        return singleton(Member::class)->summaryFields() + [
+            'methodCount' => _t(__CLASS__ . '.COLUMN_METHOD_COUNT', 'Number of registered methods'),
+            'registeredMethods' => _t(__CLASS__ . '.COLUMN_METHODS_REGISTERED', 'Registered methods'),
+            'defaultMethodName' => _t(__CLASS__ . '.COLUMN_METHOD_DEFAULT', 'Default method'),
+            'skippedRegistration' => _t(__CLASS__ . '.COLUMN_SKIPPED_REGISTRATION', 'Skipped registration'),
+        ];
+    }
+
+    public function parameterFields()
+    {
+        $methods = $this->getMethodClassToTitleMapping();
+        unset($methods[get_class(Injector::inst()->get(MethodRegistry::class)->getBackupMethod())]);
+        return FieldList::create(
+            TextField::create(
+                'Member',
+                singleton(Member::class)->i18n_singular_name()
+            )->setDescription(_t(
+                __CLASS__ . '.FILTER_MEMBER_DESCRIPTION',
+                'Firstname, Surname, Email partial match search'
+            )),
+            NumericField::create(
+                'Count',
+                _t(__CLASS__ . '.COLUMN_METHOD_COUNT', 'Number of registered methods')
+            )->setHTML5(true),
+            DropdownField::create(
+                'Methods',
+                _t(__CLASS__ . '.COLUMN_METHODS_REGISTERED', 'Registered methods'),
+                $methods,
+                ''
+            )->setHasEmptyDefault(true),
+            CheckboxField::create(
+                'Skipped',
+                _t(__CLASS__ . '.COLUMN_SKIPPED_REGISTRATION', 'Skipped registration')
+            )
+        );
+    }
+
+    protected function getMethodClassToTitleMapping()
+    {
+        $methods = array_flip(ClassInfo::implementorsOf(MethodInterface::class));
+        array_walk($methods, function (&$name, $class) {
+            $name = Injector::inst()->get($class)->getName();
+        });
+        return $methods;
+    }
+
+    /**
+     * Applies parameters to source records for filtering purposes.
+     *
+     * @param SS_List $params
```

Comment:
You can put this hinting in the array method too - although `SS_List` is too broad. Did you want to use `Filterable` instead?

Reply:
Which `array` method?

Reply:
Sorry I edited my comment just before you replied (it didn't make sense before). I meant you can use PHP7 type-hints in the method signature :)

Reply:
Yeah cool, I'll do that too, since the API is 'invisible' as I put it, there's no parent class method restricting what we can do here.